diff -u a/src/arm/unwind_i.h b/src/arm/unwind_i.h
--- a/src/arm/unwind_i.h	2011-09-11 15:06:41.000000000 +1200
+++ b/src/arm/unwind_i.h	2012-10-05 13:44:16.405047622 +1300
@@ -27,6 +27,23 @@
 
 #include <memory.h>
 #include <stdint.h>
+
+#define R0 0
+#define R1 1
+#define R2 2
+#define R3 3
+#define R4 4
+#define R5 5
+#define R6 6
+#define R7 7
+#define R8 8
+#define R9 9
+#define R10 10
+#define R11 11
+#define R12 12
+#define R13 13
+#define R14 14
+#define R15 15
 
 #include <libunwind-arm.h>
 
