--- a/configure	2012-07-23 15:24:14.469539453 +1200
+++ b/configure	2012-07-23 15:31:06.541765891 +1200
@@ -5133,7 +5133,7 @@
              | head -n 1 \
              | $SED 's/i686-apple-darwin10//' \
              | $SED 's/i686-apple-darwin11//' \
-             | $SED 's/^[^0-9]*\([0-9.]*\).*$/\1/'`
+             | $SED 's/^.* \([0-9\.]*\)/\1/g'`
 
 is_clang="notclang"
 if test "x`${CC} --version | head -n 1 | $SED 's/\(clang\) version.*/\1/'`" = "xclang" ; then
@@ -6392,7 +6392,7 @@
 	DEFAULT_SUPP="glibc-2.34567-NPTL-helgrind.supp ${DEFAULT_SUPP}"
 	DEFAULT_SUPP="glibc-2.X-drd.supp ${DEFAULT_SUPP}"
 	;;
-     2.14)
+     2.14|2.15)
 	{ $as_echo "$as_me:${as_lineno-$LINENO}: result: 2.14 family" >&5
 $as_echo "2.14 family" >&6; }
 
